### PR TITLE
fix parent_id hidden field issue

### PR DESCRIPTION
**Pull request by muhammadshoaib0k0**

### CODE REVIEW

**Comment by ShehrozIrfan** [comment]
@muhammadshoaib0k0 
Please create a short video, demonstrating the flow.

**Comment by muhammadshoaib0k0** [comment]
> @muhammadshoaib0k0 Please create a short video, demonstrating the flow.
https://www.loom.com/share/a8789379f3bb442881d2ff3ba1a14bf5

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -56,13 +56,19 @@ def show_reply_modal
   end
 
   def reply
-    @response = current_user.comments.new(reply_params[:comment])
-    if @response.save
-      flash[:success] = "Reply added successfully."
-      redirect_to comment_path(reply_params[:comment][:parent_id])
+    @response = current_user.comments.build(reply_params[:comment])
+  
+    # Check if the parent ID from the submitted form data matches the ID from the referrer URL
```
Comment (ShehrozIrfan) [inline app/controllers/comments_controller.rb:61]: remove this comment.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -56,13 +56,19 @@ def show_reply_modal
...
+    if params[:data][:comment][:parent_id] != request.referrer.split("/").last
+      # Redirect to the parent comment page with an error message if the parent ID is invalid
```
Comment (ShehrozIrfan) [inline app/controllers/comments_controller.rb:63]: this one also.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -56,13 +56,19 @@ def show_reply_modal
...
+      redirect_to comment_path(request.referrer.split("/").last), alert: "Reply does not belong to that comment id:'#{request.referrer.split("/").last}'"
     else
-      respond_to do |format|
-        format.json { render json: {errors: @response.errors.full_messages} }
+      # Save the response and redirect to the parent comment page with a success message
```
Comment (ShehrozIrfan) [inline app/controllers/comments_controller.rb:66]: this one also

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -56,13 +56,19 @@ def show_reply_modal
...
+      if @response.save
+        redirect_to comment_path(reply_params[:comment][:parent_id]), notice: "Reply added successfully"
+      else
+        # Return a JSON response with errors if the response could not be saved
```
Comment (ShehrozIrfan) [inline app/controllers/comments_controller.rb:70]: this one also

**Review by ShehrozIrfan — CHANGES_REQUESTED** [review]
Good catch.

Just a minor update, please remove all the comments added.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (muhammadshoaib0k0) [reply app/controllers/comments_controller.rb]: Done

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (muhammadshoaib0k0) [reply app/controllers/comments_controller.rb]: Done

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (muhammadshoaib0k0) [reply app/controllers/comments_controller.rb]: Done

File: app/controllers/comments_controller.rb (hunk 4 above)
Comment (muhammadshoaib0k0) [reply app/controllers/comments_controller.rb]: Done

**Review by ShehrozIrfan — APPROVED** [review]
LGTM.

Thanks for your contribution 🚀

@muhammadshoaib0k0 You can follow me and star this repo for further updates.
Thanks.

**Pull request merged**